src/PrestaShopBundle/Controller/Admin/Improve/Design/ThemeCatalogController.php

```diff
@@ -43,21 +43,7 @@ class ThemeCatalogController extends FrameworkBundleAdminController
      */
     public function indexAction(Request $request)
     {
-        $configuration = $this->get('prestashop.adapter.legacy.configuration');
-        $versionHelper = $this->get('prestashop.core.foundation.version');
-
-        $pageContent = file_get_contents(
-            'https://addons.prestashop.com/iframe/search-' . $versionHelper->getMajorVersion() . '.php?'
-            . http_build_query([
-                'psVersion' => $versionHelper->getVersion(),
-                'isoLang' => $this->getContext()->language->iso_code,
-                'isoCurrency' => $this->getContext()->currency->iso_code,
-                'isoCountry' => $this->getContext()->country->iso_code,
-                'activity' => $configuration->getInt('PS_SHOP_ACTIVITY'),
-                'parentUrl' => $request->getSchemeAndHttpHost(),
```

Comment:
looks like a good idea, but url was malformed.

Reply:
for the record @PierreRambaud, Addons platform does't expect a well formated parameter value for `parentUrl`.

For instance, let's say parentUrl value is `https://your-awesome-website.com`, then you expect to send it this way:

```
https://addons.prestashop.com/iframe/search-17.php?parentUrl=https%3A%2F%2Fyour-awesome-website.com&...
```

But the **only** valid url which works for addons is this one:

```
https://addons.prestashop.com/iframe/search-17.php?parentUrl=https://your-awesome-website.com&...
```

This is why I've give up to use the correct way to generate url and RFC-compliant urls here.

I hope you'll better understand the issue and will validate this pull request :+1:

Reply:
Weird, really weird -_-'